compiler/modules/types/table.bal

```diff
@@ -0,0 +1,71 @@
+// Implementation specific to basic type table.
+
+public type RwTableSubtype record {|
+    Bdd ro;
+    Bdd rw;
+|};
+
+public function tableContaining(SemType memberType) returns SemType {
+    Bdd ro = <Bdd> subtypeData(memberType, UT_MAPPING_RO);
+    Bdd rw = <Bdd> subtypeData(memberType, UT_MAPPING_RW);
+    SubtypeData rwt = createRwTableSubtype(ro, rw);
+    return createComplexSemType(0, [[UT_TABLE_RO, ro], [UT_TABLE_RW, rwt]]);
```

Comment:
You have to handle the cases where `ro` and `rwt` are true or false. In these cases, the ComplexSemType does not have a SubtypeData member for them (just bits in all).